### PR TITLE
feat(snapshots): encode restore spec in backup cr for restore

### PR DESCRIPTION
**Pull request by emosbaugh**
#### What this PR does / why we need it:

In order to restore the backup we need to be able to associate a restore spec with a backup.

#### Does this PR require a release note?

```release-note

```

### CODE REVIEW

File: pkg/kotsadmsnapshot/backup.go [hunk 1]
```diff
@@ -545,11 +574,8 @@ func getAppInstanceBackupSpec(k8sClient kubernetes.Interface, metadata instanceB
 			return nil, errors.New("cannot create backup for Embedded Cluster with multiple apps")
 		}
 
-		if appMeta.kotsKinds.Backup == nil {
```
Comment (emosbaugh) [inline pkg/kotsadmsnapshot/backup.go:548]: we check if this is nil on line 540

File: pkg/kotsadmsnapshot/backup.go [hunk 2]
```diff
@@ -236,7 +239,7 @@ func CreateInstanceBackup(ctx context.Context, cluster *downstreamtypes.Downstre
 
 	ctrlClient, err := k8sutil.GetKubeClient(ctx)
 	if err != nil {
-		return "", fmt.Errorf("failed to get kubeclient: %w", err)
```
Comment (emosbaugh) [inline pkg/kotsadmsnapshot/backup.go:239]: this file mostly has errors.Wrap. Making this consistent.

File: pkg/kotsadmsnapshot/backup.go [hunk 3]
```diff
@@ -324,6 +327,31 @@ func GetInstanceBackupCount(veleroBackup velerov1.Backup) int {
 	return 1
 }
 
+// GetInstanceBackupCount returns the expected number of backups from the velero backup object
+// annotation.
+func GetInstanceBackupResore(veleroBackup velerov1.Backup) (*velerov1.Restore, error) {
```
Comment (sgalsaleh) [inline pkg/kotsadmsnapshot/backup.go:332]: typo
```suggestion
func GetInstanceBackupRestore(veleroBackup velerov1.Backup) (*velerov1.Restore, error) {
```

File: pkg/kotsadmsnapshot/backup.go [hunk 4]
```diff
@@ -324,6 +327,31 @@ func GetInstanceBackupCount(veleroBackup velerov1.Backup) int {
 	return 1
 }
 
+// GetInstanceBackupCount returns the expected number of backups from the velero backup object
```
Comment (sgalsaleh) [inline pkg/kotsadmsnapshot/backup.go:330]: this description looks wrong

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**